### PR TITLE
obs-tuna: init at 1.9.6

### PR DESCRIPTION
**Pull request by shortcord**
###### Description of changes

https://github.com/univrsal/tuna

Plugin for OBS Studio.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/obs-studio/plugins/default.nix [hunk 1]
```diff
@@ -57,4 +57,6 @@
   obs-websocket = qt6Packages.callPackage ./obs-websocket.nix { }; # Websocket 4.x compatibility for OBS Studio 28+
 
   wlrobs = callPackage ./wlrobs.nix { };
+
+  obs-tuna = qt6Packages.callPackage ./obs-tuna { };
```
Comment (Lord-Valen) [inline pkgs/applications/video/obs-studio/plugins/default.nix:61]: Please respect alphabetical order.

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, obs-studio, cmake, zlib, curl, taglib, dbus
```
Comment (Lord-Valen) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, obs-studio
, cmake
, zlib
, curl
, taglib
, dbus
, pkgconfig
, qtbase
, wrapQtAppsHook
}:
```

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+, pkgconfig, qtbase, wrapQtAppsHook }:
```
Comment (Lord-Valen) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:2]:
```suggestion
```

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    mv $out/data $out/share/obs
+  '';
+
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Song information plugin for obs-studio";
+    homepage = "https://github.com/univrsal/tuna";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ shortcord ];
```
Comment (Lord-Valen) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:32]: Please add yourself to the maintainers list: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    rm -rf $out/obs-plugins
+    mv $out/data $out/share/obs
+  '';
+
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Song information plugin for obs-studio";
+    homepage = "https://github.com/univrsal/tuna";
+    license = licenses.gpl2;
```
Comment (Lord-Valen) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:31]:
```suggestion
    license = licenses.gpl2Only;
```

**Review by Lord-Valen — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix (hunk 4 above)
Comment (Lord-Valen) [reply pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:32]: This means there should essentially be two commits in this order:

1. `maintainers: add shortcord`
2. `obs-tuna: init at 1.9.6`

**Comment by Lord-Valen** [comment]
Result of `nixpkgs-review pr 237813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>obs-studio-plugins.obs-tuna</li>
  </ul>
</details>

**Review by Lord-Valen — APPROVED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+, zlib
+, curl
+, taglib
+, dbus
+, pkg-config
+, qtbase
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:15]: I think we should replace rec with the finalAttrs pattern. See an implementation here: https://github.com/NixOS/nixpkgs/pull/237742

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thank you for your first contribution.

The pr is good as it is but it would be super nice if you could make a small change and implement the finalAttrs pattern. I've linked a PR in my comment, please get inspired by it.

If you could also get rid of the `with lib;` it could be nice, just look at the linked pr and you'll understand.

Let me know if you need assistance.

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  version = "1.9.6";
+
+  nativeBuildInputs = [ cmake qtbase ];
+  buildInputs = [ obs-studio zlib curl taglib dbus pkg-config wrapQtAppsHook ];
+
+  src = fetchFromGitHub {
+    owner = "univrsal";
+    repo = "tuna";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-+AgRaivvYhogX4CLGK2ylvE8tQoauC/UMvXK6W0Tvog=";
```
Comment (drupol) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:26]:
```suggestion
    hash = "sha256-+AgRaivvYhogX4CLGK2ylvE8tQoauC/UMvXK6W0Tvog=";
```

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+, qtbase
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "obs-tuna";
+  version = "1.9.6";
+
+  nativeBuildInputs = [ cmake qtbase ];
+  buildInputs = [ obs-studio zlib curl taglib dbus pkg-config wrapQtAppsHook ];
```
Comment (drupol) [inline pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:20]: are you sure about these ? Shouldn't they belong to nativebuildinputs ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix (hunk 8 above)
Comment (shortcord) [reply pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix:20]: I was actually not sure of the differences until now, I believe a more correct solution would be the following?
```nix
nativeBuildInputs = [ cmake pkg-config wrapQtAppsHook ];
buildInputs = [ obs-studio qtbase zlib curl taglib dbus ];
```

Comment (drupol) [reply pkgs/applications/video/obs-studio/plugins/obs-tuna/default.nix]: I'm not sure yet, I'll ask to a mentor.

**Review by drupol — CHANGES_REQUESTED** [review]
Could you please:

1. Fix the merge conflict?
2. Make sure this PR contains only 2 commits:
  - One for adding the new maintainer
  - One for adding the new derivation

Let me know if you need some assistance!

**Review by fpletz — APPROVED** [review]
Tested and works. Thanks!

**Comment by drupol** [comment]
@shortcord Do you plan to update the PR ?

I'll set it in draft in the meantime. Let me know if you need some assistance.

If you don't feel like working on this anymore, please close the PR.

**Comment by shortcord** [comment]
> @shortcord Do you plan to update the PR ?
> 
> I'll set it in draft in the meantime. Let me know if you need some assistance.
> 
> If you don't feel like working on this anymore, please close the PR.

I will be updating it yes, I'll move it back to open once done.
Thanks.

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Thanks for fixing the PR !

**Pull request merged**